### PR TITLE
Make the DSL-connector rendering OCCI compliant

### PR DESCRIPTION
**Pull request by MaelAudren**
The input and output of the microservice is now OCCI compliant.
Some features are not available yet. So some parameters must be empty.

### CODE REVIEW

File: src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java [hunk 1]
```diff
@@ -44,14 +47,14 @@
 
     public ComputeBuilder() {
         this.url = "";
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java]: why only the url have "this" ?

File: src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java [hunk 2]
```diff
@@ -144,39 +147,79 @@ public ComputeBuilder state(ComputeState state) {
...
+            case STOPPED_STATE:
+                return ComputeState.SUSPENDED;
+            case PENDING_STATE:
+                return ComputeState.INACTIVE;
+            case TERMINATED_STATE:
+                return ComputeState.INACTIVE;
+            case ERROR_STATE:
+                return ComputeState.ERROR;
+        }
+        return null;
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java]: can you return Optional.absent() here?

File: src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java [hunk 3]
```diff
@@ -144,39 +147,79 @@ public ComputeBuilder state(ComputeState state) {
...
         return this;
     }
 
+    /**
+     * Set the builder according to the cloud automation model information
+     *
+     * @param cloudAutomation is the instance of the cloud automation model for a compute
+     * @return a compute builder with the cloud automation model information mapped into the builder
+     */
     public ComputeBuilder update(Model cloudAutomation) {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java]: Is it not better if it s a ComputeBuilder constructor ?

File: src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java [hunk 4]
```diff
@@ -144,39 +147,79 @@ public ComputeBuilder state(ComputeState state) {
...
+        return null;
+    }
+
+    /**
+     * Set the builder according to the resource rendering information
+     *
+     * @param rendering is the instance of the cloud automation model for a compute
+     * @return a compute builder with the cloud automation model information mapped into the builder
+     */
+    public ComputeBuilder update(ResourceRendering rendering) throws NumberFormatException {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java]: Is it not better if it s a ComputeBuilder constructor ?

File: src/main/java/org/ow2/proactive/procci/model/occi/metamodel/Entity.java [hunk 5]
```diff
@@ -44,20 +46,16 @@
  */
 @ToString
 @EqualsAndHashCode(of = {"id"})
+@Getter
 public abstract class Entity {
-    @Getter
     private String id;
-    @Getter
-    @Setter(AccessLevel.PROTECTED)
     private String title;
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/model/occi/metamodel/Entity.java]: why not final?

File: src/main/java/org/ow2/proactive/procci/model/occi/metamodel/Entity.java [hunk 6]
```diff
@@ -44,20 +46,16 @@
  */
 @ToString
 @EqualsAndHashCode(of = {"id"})
+@Getter
 public abstract class Entity {
-    @Getter
     private String id;
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/model/occi/metamodel/Entity.java]: why not final?

File: src/main/java/org/ow2/proactive/procci/model/occi/metamodel/Entity.java [hunk 7]
```diff
@@ -44,20 +46,16 @@
...
-    @Getter
-    @Setter(AccessLevel.PROTECTED)
     private String title;
-    @Getter(AccessLevel.PROTECTED)
     private final Kind kind;
-    @Getter(AccessLevel.PROTECTED)
     private List<Mixin> mixins;
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/model/occi/metamodel/Entity.java]: why not final?

File: src/main/java/org/ow2/proactive/procci/model/occi/metamodel/rendering/LinkRendering.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+                          String id, LinkLocationRendering source, LinkLocationRendering target) {
+        super(kind, mixins, attributes, actions, id);
+        this.source = source;
+        this.target = target;
+    }
+
+    public static class Builder {
+        private final String kind;
+        private List<String> mixins;
+        private Map<String, Object> attributes;
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/model/occi/metamodel/rendering/LinkRendering.java]: why not final?

File: src/main/java/org/ow2/proactive/procci/model/occi/metamodel/rendering/LinkRendering.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    private LinkRendering(String kind, List<String> mixins, Map<String, Object> attributes, List<String> actions,
+                          String id, LinkLocationRendering source, LinkLocationRendering target) {
+        super(kind, mixins, attributes, actions, id);
+        this.source = source;
+        this.target = target;
+    }
+
+    public static class Builder {
+        private final String kind;
+        private List<String> mixins;
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/model/occi/metamodel/rendering/LinkRendering.java]: why not final?

File: src/main/java/org/ow2/proactive/procci/model/occi/metamodel/rendering/LinkRendering.java (hunk 8 above)
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/model/occi/metamodel/rendering/LinkRendering.java]: why not final?

File: src/main/java/org/ow2/proactive/procci/model/occi/metamodel/rendering/LinkRendering.java [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        private Map<String, Object> attributes;
+        private List<String> actions;
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/model/occi/metamodel/rendering/LinkRendering.java]: why not final?

File: src/main/java/org/ow2/proactive/procci/request/CloudAutomationRequest.java [hunk 11]
```diff
@@ -70,14 +71,29 @@ public JSONObject getRequest() throws CloudAutomationException {
      * @throws CloudAutomationException is thrown if an error occur during the connection with CAS or the login
      */
     @Autowired
-    public Model getRequestById(String id) throws CloudAutomationException {
+    public Model getInstanceById(String id) throws CloudAutomationException {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/request/CloudAutomationRequest.java]: @Autowired on a getter method? how does it work?

File: src/main/java/org/ow2/proactive/procci/request/CloudAutomationRequest.java [hunk 12]
```diff
@@ -70,14 +71,29 @@ public JSONObject getRequest() throws CloudAutomationException {
...
+
+    /**
+     * Get the instance of cloud automation service and return the first occurance with variableValue matching variableName
+     *
+     * @param variableName  a key in variables
+     * @param variableValue the value matching with the variableName key
+     * @return the first occurance which match with variablename and variableValue
+     */
+    @Autowired
+    public Model getInstanceByVariable(String variableName, String variableValue) throws CloudAutomationException {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/request/CloudAutomationRequest.java]: @Autowired on a getter method? how does it work?

File: src/main/java/org/ow2/proactive/procci/request/CloudAutomationRequest.java [hunk 13]
```diff
@@ -70,14 +71,29 @@ public JSONObject getRequest() throws CloudAutomationException {
...
+        JSONObject instances = getRequest();
+        for (Object key : instances.keySet()) {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/request/CloudAutomationRequest.java]: lambda to clean?

File: src/main/java/org/ow2/proactive/procci/request/CloudAutomationRequest.java [hunk 14]
```diff
@@ -70,14 +71,29 @@ public JSONObject getRequest() throws CloudAutomationException {
...
+            JSONObject instanceVariables = (JSONObject) ((JSONObject) instances.get(key)).get(ModelConstant.VARIABLES);
+            if (instanceVariables.containsKey(variableName)
+                    && variableValue.equals(instanceVariables.get(variableName))) {
+                return new Model((JSONObject) instances.get(key));
+            }
         }
+        return null;
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/request/CloudAutomationRequest.java]: Optional.absent() ?

File: src/main/java/org/ow2/proactive/procci/rest/ComputeRest.java [hunk 15]
```diff
@@ -85,19 +90,18 @@
...
 
     @RequestMapping(value = "{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<Compute> getCompute(@PathVariable("id") String id) {
+    public ResponseEntity<ResourceRendering> getCompute(@PathVariable("id") String id) {
         logger.debug("Get Compute ");
         try {
-            Model computeModel = new CloudAutomationRequest().getRequestById(id);
-            if(computeModel == null){
+            Model computeModel = new CloudAutomationRequest().getInstanceByVariable(ID_NAME, id);
+            if (computeModel == null) {
```
Comment (marcocast) [inline src/main/java/org/ow2/proactive/procci/rest/ComputeRest.java]: can be improved if using optionals

File: src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java [hunk 16]
```diff
@@ -44,14 +47,14 @@
...
-        title = "";
+        title = "Compute";
         summary = "";
         architecture = null;
         cores = null;
         share = null;
         hostname = "";
         memory = null;
-        state = null;
+        state = ComputeState.ERROR;
```
Comment (MaelAudren) [inline src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java]: The state should be set each time, if the state is not set then there is an error.

File: src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java (hunk 1 above)
Comment (MaelAudren) [inline src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java]: resolved

File: src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java (hunk 16 above)
Comment (MaelAudren) [inline src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java]: Change for Optional.empty()

File: src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java [hunk 17]
```diff
@@ -144,39 +147,79 @@ public ComputeBuilder state(ComputeState state) {
 
     public ComputeBuilder state(String state) {
         switch (state) {
```
Comment (MaelAudren) [inline src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java]: default case is now Optional.empty()

File: src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java (hunk 3 above)
Comment (MaelAudren) [inline src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java]: Put as constructor

File: src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java (hunk 2 above)
Comment (MaelAudren) [inline src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java]: Improve the parsing by throwing error or returning Optional.empty()

File: src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java (hunk 4 above)
Comment (MaelAudren) [inline src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java]: Put as a constructor

**Comment by ShatalovYaroslav** [comment]
LGTM, maybe code formatting according to https://github.com/ow2-proactive/coding-rules

**Pull request merged**